x-pack/plugins/grokdebugger/public/components/grok_debugger/grok_debugger.js

```diff
@@ -82,10 +83,16 @@ export class GrokDebugger extends React.Component {
       });
 
       if (!isEmpty(simulateResponse.error)) {
-        toastNotifications.addDanger(simulateResponse.error);
+        notifications.toasts.danger({
+          body: simulateResponse.error,
+          toastLifeTimeMs: 4 * 1000,
```

Comment:
Do we need to add these lifetime amounts? I don't see them in the original code and they just raised my eyebrows.

Reply:
With the NP notifications service they seem to be necessary. Without supplying one the toasts appear and then disappear almost immediately:

![error](https://user-images.githubusercontent.com/471693/77911773-4c098600-7289-11ea-8a00-e5ab278419be.gif)

I must admit I never dug into this with metrics / logs, I just added a sensible amount of time to display. 

Happy to tweak the timeout though if needed.

Reply:
Hmm, had a little dig and I see it should be using `this.uiSettings.get('notifications:lifetime:info')` as the value if not specified, which should be `5000` by default. I'll have a look why that gets ignored.

Reply:
Okay, it turns out there's a subtle bug with the `notifications` helper that the `useKibana` hook provides (this provides things like `danger` as a shortcut with some predefined properties, but it incorrectly overrides the toast's service `toastLifeTimeMs` default if a value isn't supplied). I've changed this so that the notifications service is accessed directly, rather than the helper wrapper, and `addDanger` is called. This ensures `this.uiSettings.get('notifications:lifetime:warning')` is used for the value behind the scenes.

I'll follow this up with the relevant people on the `useKibana` hook side, but this is fixed from the Grok Debugger side here.